test/utils/compute/compute.go

```diff
@@ -0,0 +1,101 @@
+package compute
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	computev1 "go.goms.io/fleet/apis/protos/azure/compute/v1"
+)
+
+const (
+	MockAttributeBasedVMSizeRecommenderResponse = `{
+					"recommendedVmSizes": {
+						"regularVmSizes": [
+							{
+								"family": "Dsv3",
+								"name": "Standard_D2s_v3",
+								"size": "D2"
+							},
+							{
+								"family": "Standard",
+								"name": "Standard_B1s",
+								"size": "Standard_B1s"
+							}
+						]
+					}
+				}`
+)
+
+// GenerateMockAttributeBasedVMSizeRecommenderRequest is a helper function to create a mock request.
+func GenerateMockAttributeBasedVMSizeRecommenderRequest(subscriptionID, location, sku string, targetCapacity uint32) *computev1.GenerateAttributeBasedRecommendationsRequest {
+	return &computev1.GenerateAttributeBasedRecommendationsRequest{
+		SubscriptionId: subscriptionID,
+		Location:       location,
+		RegularPriorityProfile: &computev1.RegularPriorityProfile{
+			CapacityUnitType: computev1.CapacityUnitType_CAPACITY_UNIT_TYPE_VM_INSTANCE_COUNT,
+			TargetCapacity:   targetCapacity,
+		},
+		RecommendationProperties: &computev1.RecommendationProperties{
+			RestrictionsFilter: computev1.RecommendationProperties_RESTRICTIONS_FILTER_QUOTA_AND_OFFER_RESTRICTIONS,
+		},
+		ResourceProperties: &computev1.ResourceProperties{
+			VmAttributes: &computev1.VMAttributes{
+				AllowedVmSizes: []string{sku},
+			},
+		},
+	}
+}
+
+// CreateMockAttributeBasedVMSizeRecommenderServer is a helper function to create a mock server with a generated response.
+func CreateMockAttributeBasedVMSizeRecommenderServer(t *testing.T, request *computev1.GenerateAttributeBasedRecommendationsRequest, response string, httpStatusCode int) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Verify request method
+		if r.Method != http.MethodPost {
+			t.Errorf("got %s, want POST request", r.Method)
+		}
+
+		// Verify headers
+		if r.Header.Get("Content-Type") != "application/json" {
```

Comment:
```suggestion
		if r.Header.Get("Content-Type") != "application/json" {
```
use the const defined in the httputil package instead